### ISSUE TITLE
diverging history in 5 node cluster

### ISSUE THREAD

**Issue by domsj** [openvstorage/arakoon#212]
used the following commands to investigate
```
for all nodes in the cluster:

  scp nuvolat@172.17.16.8:/mnt/ssd1/arakoon/ny1-hddbackend02-abm/tlogs/* ./abm-rewind/8/
  scp nuvolat@172.17.16.8:/mnt/ssd1/arakoon/ny1-hddbackend02-abm/tlogs/head.db ./abm-rewind/8/


  ../arakoon/arakoon.native --dump-store ./8/head.db


  ../arakoon/arakoon.native -load-plugin /home/jan/projects/alba/ocaml/albamgr_plugin.cmxs --replay-tlogs ./8/ ./8/ ./6/head.db 144415000


  ../arakoon/arakoon.native --inspect-store ./8/head.db -max_results 1_000_000  | cut -d':' -f1 --complement > 8.dump


compare *.dump -> used ediff-buffers functionality of emacs for this

grep -c alba/ns/osds *.dump
grep -c alba/ns/by_id *.dump
grep -c alba/work/ *.dump
grep -c alba/preset/ *.dump
```

some results
```
grep -c alba/ns/by_id *.dump
4.dump:652
5.dump:706
6.dump:652
7.dump:706
8.dump:706
```

cause is currently unknown. ops claims to not have done manual interference.

**Comment by domsj** [comment]
similar (although less grave) issue seen on ny1-hddbackend01-abm. will check other clusters too.

**Comment by domsj** [comment]
Should be fixed by #220, in particular https://github.com/openvstorage/arakoon/commit/ff786ba07f300863b5281ee1b86f0cf662105a95#diff-593b337efe8fc715da4e2cd2dbe38814R128
In short: when catchup fails for some reason a node might apply a tlog entry the cluster will never reach consensus on.
Due to the optimistic concurrency control used by the albamgr (e.g. https://github.com/openvstorage/alba/blob/1.3.25/ocaml/src/albamgr_plugin.ml#L1528-L1532) a single different tlog entry might result in more & more differences being accumulated over time.
